lib/spack/spack/schema/packages.py

```diff
@@ -82,6 +82,23 @@
                             },
                         },
                     },
+                    # If 'get_full_repo' is promoted to a Package-level
+                    # attribute, it could be useful to set it here
+                    "set": {
+                        "type": "object",
+                        "additionalProperties": False,
+                        "properties": {
+                            "git": {
+                                "type": "string",
+                            },
+                            "url": {
+                                "type": "string",
+                            },
+                            "submodules": {
+                                "type": "boolean",
+                            },
```

Comment:
@becker33 I've updated this so you can enter arbitrary properties as strings and they are parsed as yaml values, which means that

```
x: "true"
```

is automatically applied such that `Package.x == True`. One side effect is that

```
x: "y"
```

is also converted to `Package.x == True` (since "y" is a truth value in yaml). We could allow users to fall back to something like:

```
x:
  value: "y"
  type: "string"
```

but that would remove the possibility to assign dictionary attributes using yaml structure like:

```
x:
  a: 1
  b: 2
```

although with the current logic you could do

```
x: "{a: 1, b: 2}"
```

which would automatically be loaded as a dictionary.

I'm curious if you have a preference.